exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -445,32 +445,44 @@ private static void writeScopeNameAndVersion(
   private static void writeAttributePairs(
       Writer writer, boolean initialComma, Attributes attributes) throws IOException {
     try {
-      attributes.forEach(
-          new BiConsumer<AttributeKey<?>, Object>() {
-            private boolean prefixWithComma = initialComma;
-
-            @Override
-            public void accept(AttributeKey<?> key, Object value) {
-              try {
-                if (prefixWithComma) {
-                  writer.write(',');
-                } else {
-                  prefixWithComma = true;
-                }
-                writer.write(NameSanitizer.INSTANCE.apply(key.getKey()));
-                writer.write("=\"");
-                writeEscapedLabelValue(writer, value.toString());
-                writer.write('"');
-              } catch (IOException e) {
-                throw new UncheckedIOException(e);
-              }
-            }
-          });
+      boolean prefixWithComma = initialComma;
+      for (Map.Entry<String, String> entry : sanitizeAttributePairs(attributes).entrySet()) {
```

Comment:
Typically the `.forEach` implementation is more efficient because it doesn't allocate `Map.Entry` for each iteration. See benchmark in #5183 for example.

Reply:
If we can assume it is sorted by attribute key, we might be able to check the previous key for a collision in the forEach, and append to the value rather than needing to allocate the Map.Entry.  I'll see if I can get it working.

Reply:
I think i've successfully rewritten it to use that approach.  It does rely on the keys being sorted, but it seems like that is a reasonable assumption?